ecr-login/cache/file.go

```diff
@@ -98,6 +98,9 @@ func (f *fileCredentialCache) fullFilePath() string {
 // file access. There is not guarantee here for handling multiple writes at once since there is no out of process locking.
 func (f *fileCredentialCache) save(registryCache *RegistryCache) error {
 	defer log.Flush()
+	if _, err := os.Stat(f.path); err != nil {
```

Comment:
I would have opted the other way; modifying the builder to accept a parameter defining what path to use.  That way it becomes the caller's responsibility to correctly prepare a path for use.  Shoving this in an environment variable means a caller who wants a different location than `~/.ecr` will need to override the environment variable, which seems like a really strange calling convention.